devito/ir/stree/algorithms.py

```diff
@@ -147,6 +147,12 @@ def preprocess(clusters, options=None, **kwargs):
             found = []
             for c1 in list(queue):
                 distributed_aindices = c1.halo_scheme.distributed_aindices
+                h_indices = set().union(*[(d, d.root)
+                                          for d in c1.halo_scheme.loc_indices])
+
+                # Skip if the Halo echange would end up outside its need iteration space
+                if h_indices and not h_indices & dims:
```

Comment:
Yes we do as `h_indices` can be empty (i.e no time just a single exchange) and that  would end up never processing that halo exchange since `h_indices & dims` is always false if `h_indices` is empty